### ISSUE TITLE
Sorting by type doesn't work correctly with symlinks

### ISSUE THREAD

**Issue by XPhyro** [ranger/ranger#2277]
#### Runtime Environment

- Operating system and version: Linux archlinux 5.10.23-1-lts #1 SMP Thu, 11 Mar 2021 15:32:34 +0000 x86_64 GNU/Linux
- Terminal emulator and version: st 0.8.4
- Python version: 3.9.2 [GCC 10.2.0]
- Ranger version/commit: 1.9.3
- Locale: en_GB.UTF-8

#### Current Behavior
When sorting by type, symlinks are considered regular files and thus, are not grouped together.

#### Expected Behavior
Symlinks are not considered regular files and are grouped together.

#### Possible Solutions
Group symlinks together.

#### Steps to reproduce
1. Open ranger
2. Go to a directory that has regular files and symlinks
3. Sort by type

**Comment by toonn** [comment]
I think symlinks are considered their target's type. That seems reasonable to me so this is not a fix. This'd be extra functionality imo.

**Comment by XPhyro** [comment]
Fair enough. Should I open another issue for the feature request, or keep/edit this one?

**Comment by toonn** [comment]
Feel free to edit this one.

**Comment by NoSuck** [comment]
This is an interesting idea.  Such functionality could be useful.
